### PR TITLE
feat: add masonry layout for specification blocks

### PR DESCRIPTION
**Pull request by tisidro**
- create mock masonry layout for specification blocks that will work on web browsers that do not support grid-template-rows masonry setting.

FSA22V2-319

### Spec:

Designs: [Designs](DESIGN_URL)

See Story: [FSA22V2-ISSUE](https://sparkbox.atlassian.net/browse/FSA22V2-ISSUE)

### Validation:

- [ ] This PR has code changes, and our linters still pass.
- [ ] This PR affects production code, so it was browser tested (see below).
- [ ] This PR has new code, so new tests were added or updated, and they pass.

#### To Validate:

1. Make sure all PR Checks have passed (GitHub Actions, CircleCI, Code Climate, etc).
2. Pull down all related branches.
3. Run `npm install` to make sure you have all proper dependencies. (this project requires Node 16+)
4. Copy the `.env.example` file, making sure it's in the root of your project, and rename it `.env.local`. Search for "Accessible Components" in 1Password to find the API Key and Base ID. Add both of those secrets to your newly created `.env.local` file. That should establish your connection to our Airtable database.
5. Run `npm run lint` to confirm no errors.
6. Run `npm run test` to confirm all tests pass.
7. In terminal: `npm run dev`.
8. Check [localhost:3000](http://localhost:3000/) to see changes.
9. Run axe Devtools on affected pages to confirm no urgent errors.

### Pending Questions:

1.
2.

---

#### Browser Testing

In these browsers, behavior & design closely match original specifications. A user is able to access all content and functionality, including the usability of required assistive devices, such as keyboard and screenreader.

**macOS**

- [ ] Safari (last 2 major versions)
- [ ] Chrome (last 6 months)
- [ ] Firefox (last 6 months)

**Windows**

- [ ] Chrome (last 6 months)
- [ ] Firefox (last 6 months)
- [ ] Edge (last 6 months)

**Mobile**

- [ ] Safari (last 2 major versions)
- [ ] Chrome on Android (last 6 months)
- [ ] Firefox on Android (last 6 months)

### CODE REVIEW

File: src/styles/components/_specification-block.scss [hunk 1]
```diff
@@ -13,15 +13,18 @@
     }
 
     @supports not (grid-template-rows: masonry) {
+      grid-template-rows: repeat(20, auto);
```
Comment (marissahuysentruyt) [inline src/styles/components/_specification-block.scss:16]: I think the repeating 20 is causing there to be a TON of black, empty space after the specification blocks on each page. 

Screen Shot 2022-12-06 at 1 12 06 PM

Can we just leave this as `grid-template-rows: auto`?

File: src/utils/masonryDetails.js [hunk 2]
```diff
@@ -8,41 +8,46 @@ const masonryDetails = {
...
       columnStart: 1,
     },
-    'Additional Resources': {
+    'Keyboard Interactions': {
       rowStart: 2,
+      numberOfRows: 3,
+      columnStart: 2,
+    },
+    'Additional Resources': {
+      rowStart: 3,
```
Comment (marissahuysentruyt) [inline src/utils/masonryDetails.js:48]: Each specification block, for each component, has been hard-coded with where each block is supposed to start and how many rows/columns it spans. 

Just to make sure I'm not misunderstanding- this does not use the dynamic rowStart function you showed me, correct? I hope I didn't miss it anywhere!

File: src/styles/components/_specification-block.scss (hunk 1 above)
Comment (tisidro) [reply src/styles/components/_specification-block.scss:16]: Thanks, am working on that now! :)

File: src/utils/masonryDetails.js (hunk 2 above)
Comment (tisidro) [reply src/utils/masonryDetails.js:48]: You are correct. We are doing this non-dynamically b/c we want the content blocks to always occur in the same order so it's all got to be hard-coded in.

**Review by marissahuysentruyt — APPROVED** [review]
This is it! Love it!

**Review by dustin-jw — CHANGES_REQUESTED** [review]
This looks great! I have a super nitpicky suggestion that I'm only requesting as a change because I think pushing a new commit might fix whatever issue Netlify was having with deploying this.

File: src/styles/components/_specification-block.scss [hunk 3]
```diff
@@ -13,15 +13,17 @@
     }
 
     @supports not (grid-template-rows: masonry) {
+      grid-template-rows: auto;
+
```
[the remark this answers is not in the document]
Comment (tisidro) [reply src/styles/components/_specification-block.scss]: Thanks, I just removed it. I also just tweaked the position of the "Additional Resources" on the dialog page b/c it looked crowded. Moved it from row 3 to 4 and it looks good now!

**Review by dustin-jw — DISMISSED** [review]

**Review by marissahuysentruyt — APPROVED** [review]
Looking great!

**Pull request merged**